Add -H/--hide-prompt option for accessibility

zach-is-my-name opened the pull request:
## Summary

Adds a new `-H`/`--hide-prompt` option that provides a completely clean interface by hiding both the colon prompt and cursor. This addresses accessibility needs for users with neurodivergent pathologies who find persistent visual elements distracting when reading paged content.

## Changes

- **New option**: `-H` or `--hide-prompt` 
- **Hides colon prompt**: Bottom line remains blank instead of showing `:`
- **Hides cursor completely**: Uses termcap cursor invisibility (`vi`/`ve`) with fallback to off-screen positioning
- **Preserves all functionality**: Commands still work, just without visual distractions
- **Toggle support**: Can be enabled/disabled interactively with `-H<Enter>`

## Technical Implementation

- Added `hide_prompt` boolean variable and option definition in `opttbl.c`
- Modified prompt display logic in `command.c` to conditionally show colon
- Implemented proper cursor invisibility using termcap codes (`vi`/`ve`)
- Added graceful fallback for terminals without cursor invisibility support
- Automatic cursor restoration when option is disabled

## Accessibility Benefits

This feature provides significant accessibility improvements for:
- Users with neurodivergent pathologies sensitive to visual distractions
- Anyone preferring minimal, clean interfaces for focused reading
- Environments where persistent UI elements cause cognitive interference

## Backward Compatibility

- Default behavior unchanged (option defaults to OFF)
- All existing functionality preserved
- No breaking changes to existing workflows

## Testing

Tested on macOS with various terminal emulators. The feature:
- ✅ Hides both prompt and cursor completely with `-H`
- ✅ Restores normal behavior when toggled off
- ✅ Maintains all less functionality (search, navigation, etc.)
- ✅ Gracefully handles terminals with/without cursor invisibility support

## Example Usage

```bash
# Hide prompt and cursor for distraction-free reading
less -H filename.txt

# Use long option name
less --hide-prompt filename.txt

# Toggle interactively within less
less filename.txt
# Type: -H<Enter> to hide prompt/cursor
# Type: -H<Enter> to restore prompt/cursor
```

This enhancement makes `less` more accessible while maintaining its powerful functionality.

gwsw commented:
A few initial comments:

1. With -H in effect, if you enter text in the command line at the bottom of the screen, such as via the ```/``` command, there is no cursor. This cannot be intentional, is it? For example, if you use the right/left arrow keys to move around the command line, there is no indication of where new text will be entered.
2. With -H in effect, when you exit less, the terminal remains in invisible-cursor mode when you return to the shell. This is definitely wrong. [It looks like you may have just checked in a fix for this.]
3. Are you sure it is safe to position the cursor off screen? I can't find a definitive statement about this, but the terminfo man page does say  
>A  very important point here is that the local cursor motions encoded in
       terminfo are undefined at the left and top  edges  of  a  CRT  terminal.
       Programs  should never attempt to backspace around the left edge, unless
       bw is given, and never attempt to go up locally off the top.

This is not quite the same as directly addressing an off-screen location, but it suggests that attempts to move the cursor off screen might be problematic on some terminals.
4. It looks like the vi or ve string is output at the end of every prompt. This is not terrible, but it's rather wasteful. I think it would be better to remember the cursor mode and change it only if necessary.
5. A minor point: command.c declares ```sc_move``` as an extern but never uses it, and the declarations of hide_cursor() and show_cursor() are unnecessary after rebuilding funcs.h.

zach-is-my-name commented:
Thanks for the detailed feedback\! All issues have been addressed in commit 7dd636d: cursor visibility during command input, removed unsafe off-screen positioning, optimized redundant output, and cleaned up unused declarations. The cursor restoration on exit was already fixed in commit 1994e4f.

gwsw commented:
I'm feeling that this is a large and potentially risky change for a fairly niche request. I'd suggest something like this, which will work without code changes:
```
LESS=-Ps$(tput civis | sed 's/?/\\&/') # or for most terminals just LESS=$"-Ps\e[\\?25l"
LESS_TERMCAP_cr=$(tput cnorm; tput cr)
LESSANSIENDCHARS=ml
```
This is admittedly quite kludgy, but I would prefer that the few users who need this have a few weird environment variables rather than have the 99.99+% of users who don't need it deal with a fairly risky code change.
